### ISSUE TITLE
ImageGrab.grabclipboard() for RGBA images on Windows?

### ISSUE THREAD

**Issue by teamclouday** [python-pillow/Pillow#4611]
### What did you do?
I'm trying to read RGBA image from Windows clipboard, and convert the image to RGB format with a white background. The RGBA image was copyed from chrome browser.  

### What did you expect to happen?
The returned image from ```ImageGrab.grabclipboard()``` should be in RGBA mode.  

### What actually happened?
The returned image from ```ImageGrab.grabclipboard()``` is already in RGB mode, with black background.  

### What are your OS, Python and Pillow versions?

* OS: Windows 10
* Python: 3.6.8
* Pillow: 7.0.0

```python
new_image = ImageGrab.grabclipboard()
new_image_converted = Image.new("RGB", new_image.size, (255, 255, 255))
new_image_converted.paste(new_image)
```

**Comment by teamclouday** [comment]
I think the problem comes from this line of code: https://github.com/python-pillow/Pillow/blob/a9cd55bc3990da462a5cfdb2798d706bf0057f98/src/display.c#L484  
Seems that ```CF_DIB``` does not support the alpha channel.  
Is it possible to use ```CF_DIBV5```?  
[link](https://docs.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats)

**Comment by teamclouday** [comment]
It seems that Windows automatically convert RGBA image from chrome to RGB image in clipboard.  
So it is not the problem with PIL.

**Comment by nulano** [comment]
Chrome does actually copy RGBA images properly, I can paste them just fine in Paint.NET.
Reading a few StackOverflow questions, it seems that there are multiple options for how applications store RGBA images in the clipboard:
* `CF_DIB` is supposed to be RGBX according to the docs (RGB with padding), but some applications treat it as RGBA.
* `CF_DIBV5` supports multiple formats, including RGBA (the documentation is a bit ambiguous), or even PNG.
* `'PNG'` custom format - The Windows clipboard allows you to register a custom format (based on a string). 'PNG' seems to be a common one (e.g. [Gimp](https://codereview.qt-project.org/c/qt/qtbase/+/120904/3/src/plugins/platforms/windows/qwindowsmime.cpp) or Firefox). Adding support for this should be trivial and fix this issue (with CF_DIB kept as fallback). [MSDN](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/aa359736(v%3Dvs.85)) also lists 'image/png' or 'image/x-png'.

A few links to Stack Overflow: [Win32 clipboard and alpha channel images](https://stackoverflow.com/a/15691001/1648883), [Copying From and To Clipboard loses image transparency](https://stackoverflow.com/a/46424800/1648883), [Text erased from screenshot after using Clipboard.GetImage() on Windows 10?](https://stackoverflow.com/a/46400011/1648883)

**Comment by teamclouday** [comment]
I read these as well. My problem occurred when I tried to copy plots generated by matplotlib on jupyter notebook in chrome. The workaround is to plot the graphs with full opacity.  
Chrome has no problem copying RGBA images, but information may be lost when using CF_DIB (not sure). The actual solution may be complicated.

**Comment by nulano** [comment]
Adding 'PNG' support is indeed trivial, but it seems none of Internet Explorer 11, Microsoft Edge 44, Firefox 76 or Chrome 81 actually store it in the clipboard. However, Paint.NET, and, surprisingly, MS Word 2016 do.

I also tried CF_DIBV5 (which sort-of supports RGBA) with the following results:
* Chrome stores BGRX bitplane bitmap with black background added
* Firefox stores 24-bit bitmap with black background added
* MS Edge and Internet Explorer store 24-bit bitmap, ignoring alpha
* Paint.NET stores 32-bit BGRA bitmap, which Pillow treats as BGRX (Paint.NET is abusing the format here)
* Word 2016 stores 32-bit BGRX bitplane bitmap with white background added

I then tried CF_HDROP (mentioned in the fixme comment), which is supported by both Microsoft browsers and Firefox, but not Chrome. In fact, I wrote a tool which enumerates all supported formats, and chrome only supports the three bitmap formats (I suspect it only stores CF_BITMAP and Windows converts it to the other two) and 'HTML Format' (the html image tag). Also see [the Chromium bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=963079).
